### ISSUE TITLE
Option for disabling Cinnamon spices updates

### ISSUE THREAD

**Issue by kelebek333** [linuxmint/mintupdate#684]
Hi,

Could You add an option to preferences for disabling Cinnamon spices updates?

Best regards

**Comment by clefebvre** [comment]
Why?

**Comment by rrebollo** [comment]
Maybe i am getting into something else @clefebvre but cinnamon spices updates and flatpak updates are not working behind http proxy. They seems to ignore any http proxy related setting. I am on lmde 4.

**Comment by rrebollo** [comment]
> Why?

@clefebvre Thanks for your amazing work. Could you please help me with something? I can't find where to set proxy configurations for "normal" (call from gui) mintupdate could reach spices and flatpak updates. In a terminal i set http and https proxy env vars, then call mintupdate and everything happens as expected but in any other scenario mintupdate is incapable of reach spices or flatpak updates. LMDE 4 by the way. I have been facing this trouble from previous versions but now if i forget call mintupdate in the "right" way i have to spend minutes until the process reach timeout of cinnamon spices or simply force quit of updater window.

**Comment by kelebek333** [comment]
> Hi,
> 
> Could You add an option to preferences for disabling Cinnamon spices updates?
> 
> Best regards

I disabled manually by changing CINNAMON_SUPPORT line to FALSE in mintUpdate.py .

**Comment by mtwebster** [comment]
@rrebollo Can you update spices successfully over a proxy from the individual spice pages in cinnamon settings? Mint Update should be using the same mechanism for working thru a proxy (regardless of how you start it), though I believe that mechanism expects the proxy info to be set in network settings, not via the environment.

Flatpak updates are a separate issue from spices, I'm not sure that's even attempting to use a proxy for now (we were planning on adding better support for this but it didn't make it into this release).

**Comment by IzzySoft** [comment]
> Why?

@clefebvre I'm seconding this request, and as the OP didn't outline a "Why":

* it also updates spices currently unused (suffices they've been installed once)
* it restarts Cinnamon unnecessarily for those. While no work is lost etc, this is sometimes annoying – and at least one of my spices resets on restart which I'd rather avoid
* having to "untick" them each time is not really satisfying – apart from the fact that Mintupdate then still shows pending updates
* some ppl (myself included) prefer dealing with them separately, as it was done before Uma

While on it (I'll file a separate issue if preferred), I miss the checkbox to keep the updater window open as I'd like to check the logs. This was removed with the release of Mint 20 and never came back. I've searched for a related option, but in vain.

With both, it's totally fine to have the current behavior as default, as that's most likely what the majority prefers. But the option should be there. Editing `mintupdate.py` as suggested by @kelebek333 is not really an option, as an update would overwrite that again. Personally, I'd be fine adjusting this in some text configuration file (which I couldn't yet find) – but bringing back the check-box in the update window would be more intuitive, same for a "include spices" checkk-box in the config window.

Thanks a lot for considering!

**Comment by mtwebster** [comment]
fwiw, restarting for disabled xlets should be fixed: 

https://github.com/linuxmint/mintupdate/commit/497d89cd9289b746e5cf3654c59a7e7cb8fef35d

**Comment by IzzySoft** [comment]
That's good news, thanks! But I guess it still starts the entire Cinnamon desktop if a single spice was updated, right? There might be a reason for that, and I won't argue it – but I sill prefer to update the spices manually and reload just what I updated :wink: The Cinnamon restarts I can do manually too, when needed (indicated by RAM usage and sluggishness announcing a freeze-to-come which the restart prevents).
